syntax = "proto3";

package hilo_rpc.proto;

message TestMessage {
    message Params {
        bool bool_param = 1;
        int32 int32_param = 2;
        string string_param = 3;
    }

    message Enum {
        oneof type_enum {
            bool bool_enum = 1;
            int32 int32_enum = 2;
            string string_enum = 3;
        }
    }

    Enum enum = 1;
    Params params = 2;
    map<string, string> mapping = 3;
    repeated string string_repeated = 4;
    repeated Params params_repeated = 5;
    map<int32, Params> params_map = 6;
}